model-engine/model_engine_server/infra/gateways/live_sync_model_endpoint_inference_gateway.py

```diff
@@ -186,12 +186,22 @@ async def predict(
         except UpstreamServiceError as exc:
             logger.error(f"Service error on sync task: {exc.content!r}")
             try:
+                # Try to parse traceback from the response, fallback to just return all the content if failed
                 error_json = orjson.loads(exc.content.decode("utf-8"))
-                result_traceback = (
-                    error_json.get("detail", {}).get("traceback")
-                    if isinstance(error_json, dict)
-                    else None
-                )
+                if not isinstance(error_json, dict):
```

Comment:
Wonder if we can reduce nesting

Reply:
there are 3 cases i'm checking:
detail.traceback
result.detail.traceback
result.detail[]

i could do simplification i guess